models/productDao.js

```diff
@@ -0,0 +1,61 @@
+const { teaDataSource } = require("./datasource.js");
+
+const getProduct = async (start, pageSize, category, sub, type, sort) => {
+  let whereCondition = category ? `WHERE` : type ? `WHERE` : "";
+  let andCondition = category && type ? `AND` : sub && type ? `AND` : "";
+  const categoryCondition = category ? `c.name='${category}'` : "";
+  const subCondition = sub ? `AND s.name='${sub}'` : "";
+  const typeCondition = type[0]
+    ? type[1]
+      ? type[2]
+        ? type[3]
+          ? `(t.type='${type[0]}' OR t.type='${type[1]}' OR t.type='${type[2]}',t.type='${type[3]}')`
+          : `(t.type='${type[0]}' OR t.type='${type[1]}' OR t.type='${type[2]}')`
+        : `(t.type='${type[0]}' OR t.type='${type[1]}')`
+      : `(t.type='${type[0]}')`
+    : "";
+  let sortCondition = "";
+  if (sort === "price_asc") {
+    sortCondition = `ORDER BY p.discount_price ASC`;
+  } else if (sort === "price_desc") {
+    sortCondition = `ORDER BY p.discount_price DESC`;
+  } else if (sort === "new_arrival") {
+    sortCondition = `ORDER BY p.created_at DESC`;
+  }
+  return await teaDataSource.query(
+    `SELECT
+        p.name,
+        p.price,
+        p.discount_price
+        FROM
+        products p
+        INNER JOIN subcategories s
+        ON p.subcategory_id = s.id
+        INNER JOIN categories c
+        ON s.category_id = c.id
+        INNER JOIN product_types t
+        ON p.product_types_id = t.id
+        ${whereCondition}
+        ${categoryCondition}
+        ${subCondition}
+        ${andCondition}
+        ${typeCondition}
+        ${sortCondition}
+        LIMIT ?,?`,
+    [start, pageSize]
+  );
+};
+
+const moreThanProducts = async () => {
+  return await teaDataSource.query(
+    `SELECT
+      count(*)
+    FROM
+    products`
+  );
+};
```

Comment:
제품의 전체 개수를 조회하는 쿼리문입니다.
- 우선 함수명이 함수의 행위와 예측 결과가 포함되면 좋을 것 같습니다.
- `*` 와일드카드를 사용하여 전체 컬럼을 조회하는 것보다 단순 개수를 세기 위해 id 컬럼만 추가해주는 것이 좋을 것 같습니다.
- Alias (별칭)을 통해 외부에서 해당 함수의 결과를 호출할 때 `result.counts` 등과 같이 접근할 수 있게 하면 좋을 것 같습니다.  
```suggestion
const getProductCounts = async () => {
  return await teaDataSource.query(
    `SELECT
      count(id) as counts
    FROM
    products`
  );
};
```